### ISSUE TITLE
Lightstreamer lost connection

### ISSUE THREAD

**Issue by firebird631** [ig-python/ig-markets-api-python-library#87]
I use it on IG market and I have often lost of connection, I've tryed many thing and had to modify lightstreamer.py.

First on line 385 you have self._stream_connection = None
And after if not rebind... this cannot work so I've removed it and add like this 

```
if not rebind:
	log.debug("Closing connection")
	# Clear internal data structures for session
	# and subscriptions management.
	if self._stream_connection:
		self._stream_connection.close()

	if self._session:
		self._session.clear()

	self._subscriptions.clear()
	self._current_subscription_key = 0
```

And sometime I receive often an END command from the server. So I've to constantly at a higher level to detect disconnection and reconnect. I've to do more introspection about that frequents END commands

And the disconnect method cannot work, it stay in the receive loop of the thread. So I've added a terminate boolean, set to True, to break the receive loop, else the thread never terminate.

**Comment by MohammadMoradi** [comment]
Hi @firebird631 , could you please share your code? I have the same problem. When lightsreamer lost connection no rebind is set and it stay in the loop. How can i reconnect to server without termination thread?

**Comment by firebird631** [comment]
> Hi @firebird631 , could you please share your code? I have the same problem. When lightsreamer lost connection no rebind is set and it stay in the loop. How can i reconnect to server without termination thread?

Hi @MohammadMoradi you can look on my bot project, there is a modified version with the rebind.

**Comment by oliverpolden** [comment]
Hi @firebird631, I've been having this same issue. Where can I find your bot project? You don't seem to have any public repositories.

**Comment by RRMXkun** [comment]
Fixed by PR #127

**Comment by bug-or-feature** [comment]
Closing, resolved
